### PR TITLE
go sdk blog post

### PR DESCRIPTION
**Pull request by EvanBoyle**
Blog post previewing the go sdk. closes https://github.com/pulumi/docs/issues/2513

### CODE REVIEW

File: content/blog/go-sdk-road-to-2/index.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: "Modern Cloud Infrastructure in Golang  - The Road to 2.0"
```
Comment (joeduffy) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
title: "Modern Cloud Infrastructure in Go  - The Road to 2.0"
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+authors: ["evan-boyle"]
+tags: ["go", "golang", "aws", "gcp", "azure"]
```
Comment (joeduffy) [inline content/blog/go-sdk-road-to-2/index.md]: Do we need both a `go` and `golang` tag? (Our tags are getting busy!)

File: content/blog/go-sdk-road-to-2/index.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+meta_image: "pulumigo.png"
+---
+
+Here at Pulumi, everyone on our engineering team is a Gopher. We build our open-source [pulumi/pulumi engine](https://github.com/pulumi/pulumi) and our SaaS with Go. Pulumi has supported Go in preview for more than a year, but we've recently invested in making Go support for Pulumi great as part of [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap#better-language-support" >}}).
+
+## What is Pulumi?
+
+Pulumi lets you use real languages to express your application’s infrastructure needs, using a powerful technique called “infrastructure as code.” Using infrastructure as code, you declare desired infrastructure, and an engine provisions it for you so that it’s automated, easy to replicate, and robust enough for demanding production requirements. Pulumi improves this process by leveraging real languages and making modern cloud infrastructure patterns, such as containers and serverless programs, first-class and easy.
+
+Automatically create, update, or delete cloud resources using Pulumi’s infrastructure as code engine, removing manual point-and-clicking in the cloud provider consoles, UIs, and ad-hoc scripts.
```
Comment (joeduffy) [inline content/blog/go-sdk-road-to-2/index.md]: One bit of feedback in general is that it'd be great to say here why Go specifically is a great language for infrastructure as code. This reads as a very generic overview to Pulumi and it'd be nice to emphasize Go a little bit more FWIW.

File: content/blog/go-sdk-road-to-2/index.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+title: "Modern Cloud Infrastructure in Golang  - The Road to 2.0"
+authors: ["evan-boyle"]
+tags: ["go", "golang", "aws", "gcp", "azure"]
+date: "2020-02-25"
+meta_desc: "Pulumi + Go is a powerful combo for your cloud-native infrastructure."
+meta_image: "pulumigo.png"
+---
+
+Here at Pulumi, everyone on our engineering team is a Gopher. We build our open-source [pulumi/pulumi engine](https://github.com/pulumi/pulumi) and our SaaS with Go. Pulumi has supported Go in preview for more than a year, but we've recently invested in making Go support for Pulumi great as part of [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap#better-language-support" >}}).
```
Comment (joeduffy) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
Here at Pulumi, everyone on our engineering team is a Gopher. Go has quickly become the "language of the cloud," and so when we chose to build our open-source [pulumi/pulumi engine](https://github.com/pulumi/pulumi) and SaaS backend, we chose Go. As such, we are very excited to welcome Go to the family of supported infrastructure as code languages as part of [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap#better-language-support" >}}).
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        }
+
+        // Export the name of the bucket
+        ctx.Export("bucketName", bucket.ID())
+        return nil
+    })
+}
+```
+
+## An Expressive and Idiomatic Type System
```
Comment (joeduffy) [inline content/blog/go-sdk-road-to-2/index.md]: One aspect of the Go SDK that is worth mentioning is that we designed it to give you a little more control, with the intent of being embeddable within larger systems. That's why, for instance, you get the `*pulumi.Context` object. It's true we stopped short of programmatically driving deployments (per lots of recent discussions), however we've actually seen that play out in practice with some Go-oriented customers.

File: content/blog/go-sdk-road-to-2/index.md (hunk 2 above)
Comment (lukehoban) [reply content/blog/go-sdk-road-to-2/index.md]: JFWIW - I opened https://github.com/pulumi/docs/issues/2532 just now on the general issue here.

File: content/blog/go-sdk-road-to-2/index.md (hunk 5 above)
Comment (EvanBoyle) [reply content/blog/go-sdk-road-to-2/index.md]: This seems like a topic that could benefit from a post of its own similar to the type system deep dive. Much of this is undocumented and doesn't have public prior art to reference.

The goal of this post is to drive people through to the getting started and examples pages. Do you think that adding a section on programmatic access would help to this end, or would that topic be better as a standalone with some richer content?

File: content/blog/go-sdk-road-to-2/index.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+---
+title: "Modern Cloud Infrastructure in Go - The Road to 2.0"
+authors: ["evan-boyle"]
+tags: ["go", "aws", "gcp", "azure"]
+date: "2020-02-25"
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
date: "2020-02-27"
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+Here at Pulumi, everyone on our engineering team is a Gopher. Go has quickly become the "language of the cloud," and so when we chose to build our open-source [pulumi/pulumi engine](https://github.com/pulumi/pulumi) and SaaS backend, we chose Go. As such, we are very excited to welcome Go to the family of supported infrastructure as code languages as part of [Pulumi 2.0]({{< relref "/blog/pulumi-2-0-roadmap#better-language-support" >}}).
+
+## What is Pulumi?
+
+Pulumi lets you use real languages to express your application’s infrastructure needs, using a powerful technique called “infrastructure as code.” Using infrastructure as code, you declare desired infrastructure, and an engine provisions it for you so that it’s automated, easy to replicate, and robust enough for demanding production requirements. Pulumi improves this process by leveraging real languages and making modern cloud infrastructure patterns, such as containers and serverless programs, first-class and easy.
+
+Automatically create, update, or delete cloud resources using Pulumi’s infrastructure as code engine, removing manual point-and-clicking in the cloud provider consoles, UIs, and ad-hoc scripts.
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
- Automatically create, update, or delete cloud resources using Pulumi’s infrastructure as code engine, removing manual point-and-clicking in the cloud provider consoles, UIs, and ad-hoc scripts.
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+Use your favorite IDEs and tools, including Visual Studio Code, taking advantage of features like auto-completion, refactoring, and interactive documentation.
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
- Use your favorite IDEs and tools, including Visual Studio Code, taking advantage of features like auto-completion, refactoring, and interactive documentation.
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+Catch mistakes early on with standard compiler errors, and an infrastructure-specific policy engine for enforcing security, compliance, and best practices.
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
- Catch mistakes early on with standard compiler errors, and an infrastructure-specific policy engine for enforcing security, compliance, and best practices.
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+Reuse any existing go packages, or distribute your own, whether that’s for infrastructure best practices, productivity, or just general programming patterns.
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
- Reuse any existing go packages, or distribute your own, whether that’s for infrastructure best practices, productivity, or just general programming patterns.
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+Build scalable cloud applications using classic infrastructure cloud-native technologies. including serverless functions, and highly scalable databases such as AWS Aurora, into your core development experience, bringing them closer to your application code.
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
- Build scalable cloud applications using classic infrastructure cloud-native technologies. including serverless functions, and highly scalable databases such as AWS Aurora, into your core development experience, bringing them closer to your application code.
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+
+Inputs, Outputs, and functions that operate over them ([All]({{< relref "/docs/intro/concepts/programming-model#all" >}}), [Apply]({{< relref "/docs/intro/concepts/programming-model#apply" >}}), [Sprintf]({{< relref "/docs/intro/concepts/programming-model#outputs-and-strings" >}})) must be compatible with raw type primitives (string, integer, list, map, boolean), and user-defined structs. In other words, we must support polymorphism in a language that doesn’t have generics. We hide the gritty details behind a rich, strongly-typed API that we were able to iterate on easily thanks to the utilization of [code generation](https://github.com/pulumi/pulumi/blob/master/sdk/go/pulumi/types_builtins.go).
+
+## Try Pulumi + Go Today
+
+Although Pulumi for Go is in “preview” status, it supports all of the essential Pulumi programming model features (and the rest is on its way). Our goal is to gather feedback and over the next few weeks, and work hard to improve the Go experience across the board, including more examples and better documentation. To get started check out:
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
Although Pulumi for Go is in “preview” status until 2.0 is released, it supports all of the essential Pulumi programming model features (and the rest is on its way). Our goal is to gather feedback and over the next few weeks, and work hard to improve the Go experience across the board, including more examples and better documentation. To get started check out:
```

File: content/blog/go-sdk-road-to-2/index.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+
+Automatically create, update, or delete cloud resources using Pulumi’s infrastructure as code engine, removing manual point-and-clicking in the cloud provider consoles, UIs, and ad-hoc scripts.
+Use your favorite IDEs and tools, including Visual Studio Code, taking advantage of features like auto-completion, refactoring, and interactive documentation.
+Catch mistakes early on with standard compiler errors, and an infrastructure-specific policy engine for enforcing security, compliance, and best practices.
+Reuse any existing go packages, or distribute your own, whether that’s for infrastructure best practices, productivity, or just general programming patterns.
+Build scalable cloud applications using classic infrastructure cloud-native technologies. including serverless functions, and highly scalable databases such as AWS Aurora, into your core development experience, bringing them closer to your application code.
+
+Pulumi’s free open source SDK, which includes a CLI and assortment of libraries, enables these capabilities.
+
+Creating an AWS S3 Bucket is familiar and simple:
```
Comment (spara) [inline content/blog/go-sdk-road-to-2/index.md]:
```suggestion
Creating an AWS S3 Bucket is a familiar and simple example that demonstrates these advantages:
```

File: content/blog/go-sdk-road-to-2/index.md (hunk 3 above)
Comment (EvanBoyle) [reply content/blog/go-sdk-road-to-2/index.md]: Updated.

**Pull request merged**